### PR TITLE
restored filters

### PR DESCRIPTION
**Pull request by AvKhokhlov**

### CODE REVIEW

File: BaikalStandalone/Application/app_utils.cpp [hunk 1]
```diff
@@ -187,6 +187,31 @@ namespace Baikal
             s.cmd_line_mode = true;
         }
 
+#ifdef ENABLE_DENOISER
+        if (m_cmd_parser.OptionExists("-denoiser_type"))
+        {
+            auto denoiser_mode = m_cmd_parser.GetOption("-denoiser_type");
```
Comment (dtarakanov1) [inline BaikalStandalone/Application/app_utils.cpp]: `type` or `mode`?

File: BaikalStandalone/Application/cl_render.cpp [hunk 2]
```diff
@@ -63,11 +63,25 @@ namespace Baikal
...
+        {
+            case DenoiserMode::kBilateral:
+                AddPostEffect(m_primary, PostEffectType::kBilateralDenoiser);
+                break;
+            case DenoiserMode::kWavelet:
+                AddPostEffect(m_primary, PostEffectType::kWaveletDenoiser);
+                break;
+            case DenoiserMode ::kML:
+                AddPostEffect(m_primary, PostEffectType::kMLDenoiser);
+                m_post_effect->SetParameter("gpu_memory_fraction", settings.gpu_mem_fraction);
```
Comment (achernigin1987) [inline BaikalStandalone/Application/cl_render.cpp]: I think we should set params  for all denoisers in the single place

File: BaikalStandalone/Application/app_utils.h [hunk 3]
```diff
@@ -100,6 +100,12 @@ namespace Baikal
         // device settings
         float gpu_mem_fraction = 0; // float number from 0 to 1, percentage of max used device memory, 0 for default behavior
         std::string visible_devices;
+
+#ifdef ENABLE_DENOISER
+        // denoiser settings
+        DenoiserMode denoiser_mode;
```
Comment (dtarakanov1) [inline BaikalStandalone/Application/app_utils.h]: Let's initialize in-place this field and start_spp.

File: BaikalStandalone/Application/app_utils.cpp [hunk 4]
```diff
@@ -187,6 +187,31 @@ namespace Baikal
...
+                s.denoiser_mode = DenoiserMode::kWavelet;
+            }
+            else if (denoiser_mode == "ml")
+            {
+                s.denoiser_mode = DenoiserMode::kML;
+            }
+        }
+        else
+        {
+            s.denoiser_mode = DenoiserMode::kML;
```
Comment (dtarakanov1) [inline BaikalStandalone/Application/app_utils.cpp]: Not needed as there is the default setting value.

File: BaikalStandalone/Application/application.cpp [hunk 5]
```diff
@@ -966,18 +966,18 @@ namespace Baikal
             if (m_cl->GetPostEffectType() != PostEffectType::kMLDenoiser) {
```
Comment (dtarakanov1) [inline BaikalStandalone/Application/application.cpp]: `{` to the next line.

File: BaikalStandalone/Application/application.cpp [hunk 6]
```diff
@@ -966,18 +966,18 @@ namespace Baikal
...
                 ImGui::SliderFloat("Position sigma", &sigmaPosition, 0.f, 0.3f);
                 ImGui::SliderFloat("Normal sigma", &sigmaNormal, 0.f, 5.f);
                 ImGui::SliderFloat("Color sigma", &sigmaColor, 0.f, 5.f);
 
-                if (m_cl->GetDenoiserFloatParam("position_sensitivity").x != sigmaPosition ||
-                    m_cl->GetDenoiserFloatParam("normal_sensitivity").x != sigmaNormal ||
-                    m_cl->GetDenoiserFloatParam("color_sensitivity").x != sigmaColor) {
+                if (m_cl->GetDenoiserFloatParam("position_sensitivity") != sigmaPosition ||
+                    m_cl->GetDenoiserFloatParam("normal_sensitivity") != sigmaNormal ||
+                    m_cl->GetDenoiserFloatParam("color_sensitivity") != sigmaColor) {
```
Comment (dtarakanov1) [inline BaikalStandalone/Application/application.cpp]: `{` to the next line.

File: BaikalStandalone/Application/application.cpp (hunk 5 above)
Comment (dtarakanov1) [reply BaikalStandalone/Application/application.cpp]: Up.

**Pull request merged**